src/pages/instances/InstanceSnapshots.tsx

```diff
@@ -1,30 +1,47 @@
 import React, { FC, useState } from "react";
-import { MainTable, Row } from "@canonical/react-components";
+import {
+  Button,
+  Col,
+  MainTable,
+  Row,
+  SearchBox,
+} from "@canonical/react-components";
 import { isoTimeToString } from "util/helpers";
 import DeleteSnapshotBtn from "./actions/snapshots/DeleteSnapshotBtn";
 import RestoreSnapshotBtn from "./actions/snapshots/RestoreSnapshotBtn";
-import { createPortal } from "react-dom";
-import CreateSnapshotBtn from "./actions/snapshots/CreateSnapshotBtn";
 import { LxdInstance } from "types/instance";
 import EmptyState from "components/EmptyState";
 import CreateSnapshotForm from "pages/instances/actions/snapshots/CreateSnapshotForm";
 
 interface Props {
-  controlTarget?: HTMLSpanElement | null;
   instance: LxdInstance;
 }
 
-const InstanceSnapshots: FC<Props> = ({ controlTarget, instance }) => {
+const InstanceSnapshots: FC<Props> = ({ instance }) => {
+  const [query, setQuery] = useState<string>("");
   const [isModalOpen, setModalOpen] = useState(false);
+
+  const visibleSnapshots =
```

Comment:
Better use `filteredSnapshots`. I also renamed it in the InstanceList on the pagination PR.